### PR TITLE
[16.0][ADD] stock_move_free_reservation_reassign: New Addon

### PR DESCRIPTION
**Pull request by lmignon**
Automatically try to reassign moves freed by an inventory adjsutment

### CODE REVIEW

**Comment by lmignon** [comment]
cc @rousseldenis @jbaudoux

File: stock_move_free_reservation_reassign/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Stock Move Free Reservation Reassign",
+    "summary": """
+        Try to reassign move when reservation is removed after a quant is became unavailable""",
```
Comment (jbaudoux) [inline stock_move_free_reservation_reassign/__manifest__.py]:
```suggestion
        Try to reassign a move when its reservation is removed due to the related quant becoming unavailable""",
```

**Review by jbaudoux — APPROVED** [review]
LGTM, not tested

**Comment by jbaudoux** [comment]
Missing empty README.rst

**Comment by rousseldenis** [comment]
> Missing empty README.rst

Nope, will be generated at merge

**Comment by rousseldenis** [comment]
> > Missing empty README.rst
> 
> Nope, will be generated at merge

Oh, yes, pre-commit is complaining. @lmignon

**Comment by lmignon** [comment]
@jbaudoux @rousseldenis Everything is ok now...

**Review by rousseldenis — APPROVED** [review]
Code review

File: stock_move_free_reservation_reassign/tests/test_stock_move_free_reservation_reassign.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        # by default reassign is disabled
+        self.assertEqual(self.product.qty_available, 20.0)
+        move = self._create_move_picking_out(self.product, 10.0)
+        move._action_assign()
+        self.assertEqual(move.state, "assigned")
+        # make an inventory on the reserved location
+        initial_location = move.move_line_ids.location_id
+        self._make_location_inventory(self.product, initial_location, 0)
+        self.assertEqual(self.product.qty_available, 10.0)
+        # the move should still be assigned but a new move should be created
```
Comment (TDu) [inline stock_move_free_reservation_reassign/tests/test_stock_move_free_reservation_reassign.py:104]: The comment does not fit with the assertion ?

File: stock_move_free_reservation_reassign/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        related quant becoming unavailable""",
+    "version": "16.0.1.0.0",
+    "license": "AGPL-3",
+    "author": "ACSONE SA/NV,Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/stock-logistics-workflow",
+    "depends": [
+        "stock",
+    ],
+    "data": ["views/res_config_settings.xml"],
+    "demo": [],
```
Comment (TDu) [inline stock_move_free_reservation_reassign/__manifest__.py:17]:
```suggestion
```

**Review by TDu — APPROVED** [review]
Whoua, temporary patching a function to retrieve affected records by an other function with the use of two context manager :thinking: :exploding_head: :+1:

**Pull request merged**